pkg/frontend/clustermanager_putorpatch.go

```diff
@@ -0,0 +1,142 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/database/cosmosdb"
+	"github.com/Azure/ARO-RP/pkg/frontend/middleware"
+	"github.com/Azure/ARO-RP/pkg/util/arm"
+)
+
+func (f *frontend) putOrPatchClusterManagerConfiguration(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	log := ctx.Value(middleware.ContextKeyLog).(*logrus.Entry)
+	vars := mux.Vars(r)
+	var header http.Header
+	var b []byte
+
+	if disableOCMAPI {
+		reply(log, w, nil, []byte("forbidden."), api.NewCloudError(http.StatusForbidden, api.CloudErrorCodeForbidden, "", "forbidden."))
+		return
+	}
+
+	err := cosmosdb.RetryOnPreconditionFailed(func() error {
+		var err error
+		b, err = f._putOrPatchClusterManagerConfiguration(ctx, log, r, &header, f.apis[vars["api-version"]].ClusterManagerConfigurationConverter)
+		return err
+	})
+
+	reply(log, w, header, b, err)
+}
+
+func (f *frontend) _putOrPatchClusterManagerConfiguration(ctx context.Context, log *logrus.Entry, r *http.Request, header *http.Header, converter api.ClusterManagerConfigurationConverter) ([]byte, error) {
+	body := r.Context().Value(middleware.ContextKeyBody).([]byte)
+	correlationData := r.Context().Value(middleware.ContextKeyCorrelationData).(*api.CorrelationData)
+	systemData, _ := r.Context().Value(middleware.ContextKeySystemData).(*api.SystemData) // don't panic
+
+	_, err := f.validateSubscriptionState(ctx, r.URL.Path, api.SubscriptionStateRegistered)
+	if err != nil {
+		return nil, err
+	}
+
+	originalPath := r.Context().Value(middleware.ContextKeyOriginalPath).(string)
+	armResource, err := arm.ParseArmResourceId(originalPath)
+	if err != nil {
+		return nil, err
+	}
+	clusterURL := strings.ToLower(armResource.ParentResourceToString())
```

Comment:
Rename function in armResource.  Idiomatic nit

```suggestion
	clusterURL := strings.ToLower(armResource.ParentResource())

Reply:
nit - whatever you want to do